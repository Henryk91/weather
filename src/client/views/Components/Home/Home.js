import React, { Component } from 'react';
import { DefaultDay } from '../index'

export default class Home extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        let dayBlock = null;
        const weatherData = this.props.weatherData;
        if (weatherData) {
            dayBlock = weatherData.daily.data.map((val, i) => {
                return (
                    <DefaultDay
                        key={val.time}
                        currently={weatherData.currently}
                        data={val} id={i}
                    />
                )
            })
        }
        return (
            <div className="bigScreen" id="home">
                {weatherData ?
                    <section>
                        <p>{weatherData.daily.summary}</p>
                        <p>(click day for details)</p>
                    </section> :
                    <h3>Please enter a location</h3>
                }
                <main className="weekDays"> {dayBlock ? dayBlock : null} </main>
            </div>
        );
    }
}
