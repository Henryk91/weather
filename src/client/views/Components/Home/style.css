.weekDays{
    display:grid;
    grid-template-columns:repeat(3, 1fr);
    grid-auto-rows: auto;
}
.weekDays :first-child{
    grid-column:1/4;
  }
  .weekDays :last-child{
    grid-column:2/3;
  }