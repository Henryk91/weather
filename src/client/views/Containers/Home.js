import React, { Component } from 'react';
import DefaultDay from './DefaultDay'

export default class Home extends Component {
    constructor(props) {
        super(props);
        this.state = {
            weatherData: null,
            searching: false
        };
    }

    render() {
        let dayBlock = null;
        if (this.props.weatherData) {
            dayBlock = this.props.weatherData.daily.data.map((val) => {
                return (
                    <div key={val.time} >
                        <DefaultDay data={val} />
                    </div>
                )
            })
        }

        return (
            <div className="bigScreen" id="home">
            
                {this.props.weatherData ? <div><p>{this.props.weatherData.daily.summary}</p>
                 <p>(click day for details)</p></div> : <h3>Please enter a location</h3>}
                <div className="weekDays">{dayBlock ? dayBlock : null}</div>
            </div>
        );
    }
}
