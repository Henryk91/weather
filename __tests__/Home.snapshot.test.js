
import Home from '../src/client/views/Components/Home/Home';
import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';
import { weatherData } from "../testData/weatherData.sample"

test('Home component renders with summary and defaultday', () => {

  const rendered = renderer.create(
    <Router>
                <Home weatherData={weatherData()} />
        </Router>
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});