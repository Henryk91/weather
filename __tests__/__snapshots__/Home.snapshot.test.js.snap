// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders with summary and defaultday 1`] = `
<div
  className="bigScreen"
  id="home"
>
  <section>
    <p>
      Light rain today through Tuesday, with high temperatures rising to 31°C on Thursday.
    </p>
    <p>
      (click day for details)
    </p>
  </section>
  <main
    className="weekDays"
  >
     
    <a
      href="/detailed/1552082400"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <section
        className="dayBox blueHover"
      >
        <p
          className="leadP"
        >
          Today
        </p>
        <canvas
          height={40}
          width={40}
        />
        <p>
          Current Temp: 
          29.3
          °
        </p>
        <p>
          18
          °/
          30
          °
        </p>
        <p>
          Rain: 
          23
          %
        </p>
      </section>
    </a>
    <a
      href="/detailed/1552168800"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <section
        className="dayBox blueHover"
      >
        <p
          className="leadP"
        >
          Sun Mar 10 
        </p>
        <canvas
          height={40}
          width={40}
        />
        <p>
          18
          °/
          28
          °
        </p>
        <p>
          Rain: 
          55
          %
        </p>
      </section>
    </a>
    <a
      href="/detailed/1552255200"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <section
        className="dayBox blueHover"
      >
        <p
          className="leadP"
        >
          Mon Mar 11 
        </p>
        <canvas
          height={40}
          width={40}
        />
        <p>
          19
          °/
          29
          °
        </p>
        <p>
          Rain: 
          42
          %
        </p>
      </section>
    </a>
    <a
      href="/detailed/1552341600"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <section
        className="dayBox blueHover"
      >
        <p
          className="leadP"
        >
          Tue Mar 12 
        </p>
        <canvas
          height={40}
          width={40}
        />
        <p>
          18
          °/
          30
          °
        </p>
        <p>
          Rain: 
          25
          %
        </p>
      </section>
    </a>
    <a
      href="/detailed/1552428000"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <section
        className="dayBox blueHover"
      >
        <p
          className="leadP"
        >
          Wed Mar 13 
        </p>
        <canvas
          height={40}
          width={40}
        />
        <p>
          20
          °/
          31
          °
        </p>
        <p>
          Rain: 
          20
          %
        </p>
      </section>
    </a>
    <a
      href="/detailed/1552514400"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <section
        className="dayBox blueHover"
      >
        <p
          className="leadP"
        >
          Thu Mar 14 
        </p>
        <canvas
          height={40}
          width={40}
        />
        <p>
          18
          °/
          31
          °
        </p>
        <p>
          Rain: 
          16
          %
        </p>
      </section>
    </a>
    <a
      href="/detailed/1552600800"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <section
        className="dayBox blueHover"
      >
        <p
          className="leadP"
        >
          Fri Mar 15 
        </p>
        <canvas
          height={40}
          width={40}
        />
        <p>
          17
          °/
          31
          °
        </p>
        <p>
          Rain: 
          28
          %
        </p>
      </section>
    </a>
    <a
      href="/detailed/1552687200"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <section
        className="dayBox blueHover"
      >
        <p
          className="leadP"
        >
          Sat Mar 16 
        </p>
        <canvas
          height={40}
          width={40}
        />
        <p>
          19
          °/
          30
          °
        </p>
        <p>
          Rain: 
          22
          %
        </p>
      </section>
    </a>
     
  </main>
</div>
`;
